lms/models/lti_role.py

```diff
@@ -0,0 +1,145 @@
+from enum import Enum
+
+import sqlalchemy as sa
+from sqlalchemy.ext.hybrid import hybrid_property
+
+from lms.db import BASE
+
+
+class RoleType(str, Enum):
+    """Enum for the different types of role a user can have."""
+
+    INSTRUCTOR = "instructor"
+    LEARNER = "learner"
+    ADMIN = "admin"
+
+    @classmethod
+    def parse_role(cls, role):
+        # We have to do this work around because Enums can't have private
+        # attributes. They are just interpreted as extra values for the enum.
+        return _RoleParser.parse_role(role)
+
+
+class LTIRole(BASE):
+    """Model for LTI role strings and our interpretation of them."""
+
+    __tablename__ = "lti_role"
+
+    # Export for others convenience
+    Type = RoleType
+
+    id = sa.Column(sa.Integer(), autoincrement=True, primary_key=True)
+
+    _value = sa.Column("value", sa.UnicodeText(), nullable=False, unique=True)
+    """The raw string from LTI params."""
+
+    type = sa.Column("type", sa.Enum(RoleType), nullable=False, unique=False)
+    """Our interpretation of the value."""
```

Comment:
This totally works, but it causes the type of the column in SQL to actually be "roletype" which suggests SQLAlchemy is creating some custom thing in Postgres rather than storing a string as per the Stackoverflow answer I cribbed this from...